### ISSUE TITLE
[Production] [0006] Saga: consider the best finite state machine (FSM) packaging

### ISSUE THREAD

**Issue by mwpowellhtx** [mwpowellhtx/KP-Liberation#25]
### Basic Information
**Mission version:** `0.98.dev-s2`
**Map used:** _Altis_
**Mods used:** _..._

### Short Issue Description
We have several choices. We could roll our own. We did that in draft for the legacy, unfactored code. We may consider an _FSM editor_ based approach, either the [_Arma 3 Tools_](https://store.steampowered.com/app/233800/Arma_3_Tools/) or [_FSM Edit_](https://www.armaholic.com/page.php?id=1184) based. We may also investigate a [_CBA FSM_](https://cbateam.github.io/CBA_A3/docs/files/statemachine/fnc_create-sqf.html) based approach.

### Expected Behaviour
Bottom line we like the idea of introducing a proper [_finite state machine_](https://en.wikipedia.org/wiki/Finite-state_machine) ([FSM](https://en.wikipedia.org/wiki/Finite-state_machine)), whatever the level of support, packaging, etc. We must still decide whether to adopt a one machine per sector approach, or one overall machine for all such sectors.

**Comment by mwpowellhtx** [comment]
Ran with CBA state machine, from configuration. Works great! Tools for editing may work, may not work, depending on the tool. And includes extra bits thrown in. But really, these offer minimal value for what they are. It is easy enough to visualize  the states and  their transitions once you comprehend what is going on. More important is to have in mind the goals you want the SM to accomplish.
